src/network/modules/blockchain/blockchain.ts

```diff
@@ -1,17 +1,355 @@
+  hash: ArrayBuffer
+  height: number
+}
+
+export type TransactionIdentifier = ArrayBuffer
+
+export type Transaction = {
+  id: ArrayBuffer
+  request?: {
+    from?: string
+    to?: string
+    method: string
+    timestamp: string
+    cbor: string
+  }
+  response?: {
+    from: string
+    to?: string
+    timestamp: Date
+    cbor: string
+  }
+}
+
+export type TransactionReturns = {
+  txn?: Transaction
+}
+
+export enum SingleBlockQueryType {
+  hash = "hash",
+  height = "height",
+}
+
+export enum RangeBlockQueryType {
+  height = "height",
+  time = "time",
+}
+
+export type SingleBlockQueryValueType = ArrayBuffer | Number
+
+export type Block = {
+  id: BlockIdentifier
+  parent?: BlockIdentifier
+  appHash: ArrayBuffer
+  timestamp: number
+  txnCount: number
+  txns: Transaction[]
+}
+
+export type InfoReturns = {
+  info?: {
+    latestBlock: BlockIdentifier
+    appHash?: ArrayBuffer
+    retainedHeight?: number
+  }
+}
+
+export type BlockReturns = {
+  block?: Block
+}
+
+export type BlockListReturns = {
+  latestHeight: number
+  blocks: BlockReturns[]
+}
+
+interface Blockchain extends NetworkModule {
+  info: () => Promise<InfoReturns>
+  block: (
+    queryType: SingleBlockQueryType,
+    value: SingleBlockQueryValueType,
+  ) => Promise<BlockReturns>
+  list(opts: {
+    queryType: RangeBlockQueryType
+    count?: number
+    order?: ListOrderType
+    range?: RangeBounds<number> | RangeBounds<Date>
+  }): Promise<BlockListReturns>
+  transaction: (txnHash: ArrayBuffer) => Promise<TransactionReturns>
+  request: (txnHash: ArrayBuffer) => Promise<unknown>
+  response: (txnHash: ArrayBuffer) => Promise<unknown>
+}
+
+export const Blockchain: Blockchain = {
+  _namespace_: "blockchain",
+
+  async info(): Promise<InfoReturns> {
+    const msg = await this.call("blockchain.info")
+    return await getBlockainInfo(msg)
   },
 
-  async block(): Promise<unknown> {
-    // @ts-ignore
-    return this.call("blockchain.block", {})
+  async block(
+    queryType: SingleBlockQueryType,
+    value: SingleBlockQueryValueType,
+  ): Promise<BlockReturns> {
+    const m = new Map()
+    if (queryType === "hash" && value instanceof ArrayBuffer) {
+      m.set(0, value)
+    } else if (queryType === "height" && typeof value === "number") {
+      m.set(1, value)
+    }
+    const msg = (await this.call(
+      "blockchain.block",
+      new Map().set(0, m),
+    )) as Message
+    const payload = msg.getPayload()
+    const blockData = payload.get(0)
+    return await getBlock(blockData, this.blockchain)
   },
 
-  async transaction(): Promise<unknown> {
-    // @ts-ignore
-    return await this.call("blockchain.transaction", new Map([[1, symbols]]))
+  async list({
+    queryType,
+    count = 10,
+    order = ListOrderType.descending,
+    range = [],
+  }: {
+    queryType: RangeBlockQueryType
+    count?: number
+    order?: ListOrderType
+    range?: RangeBounds<number> | RangeBounds<Date>
+  }): Promise<BlockListReturns> {
```

Comment:
Yeah, it'd be nice to have a declared interface for list options so we don't have to repeat them here.